### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by offsset**
https://offsset.github.io/react_tabs/

### CODE REVIEW

**Review by AlisaVlasova — CHANGES_REQUESTED** [review]
check your demo
404 on the tab click 
image

File: src/App.tsx [hunk 1]
```diff
@@ -10,31 +11,31 @@ export const tabs = [
 ];
 
 export const App: React.FC = () => {
+  const [selectedTabId, setSelectedTabId] = useState('tab-1');
+
+  const getselectedTabId = (id: string) => {
```
Comment (tetiana-ivanchenko) [inline src/App.tsx:16]: if your function names start with "get" the expected behavior will be if the function returns smth - it means you will get some data. Your function only sets some value for the state, rename it please

File: src/App.tsx [hunk 2]
```diff
@@ -10,31 +11,31 @@ export const tabs = [
...
+    setSelectedTabId(id);
+  };
+
+  const tabSelectText = tabs.find(tab => tab.id === selectedTabId);
```
Comment (tetiana-ivanchenko) [inline src/App.tsx:20]: also rename it, it returns not just text

File: src/components/Tabs/Tabs.tsx [hunk 3]
```diff
@@ -1 +1,43 @@
...
+
+export const Tabs: React.FC<Props> = (
+  { tabs, selectedTabId, getSelectedTabId },
+) => {
+  return (
+    <ul>
+      {
+        tabs.map(tab => {
+          const { id, title } = tab;
+          const booleanClass = id === selectedTabId;
```
Comment (tetiana-ivanchenko) [inline src/components/Tabs/Tabs.tsx:22]: follow the naming convention for the boolan variables

File: src/components/Tabs/Tabs.tsx [hunk 4]
```diff
@@ -1 +1,43 @@
...
+
+          return (
+            <li
+              className={ClassNames({ 'is-active': booleanClass })}
+              data-cy="Tab"
+            >
+              <a
+                // href={`'#' + ${id}`}
```
Comment (tetiana-ivanchenko) [inline src/components/Tabs/Tabs.tsx:30]: don't left  comments

**Review by tetiana-ivanchenko — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 5]
```diff
@@ -10,31 +11,31 @@ export const tabs = [
 ];
 
 export const App: React.FC = () => {
+  const [selectedTabId, setSelectedTabId] = useState('tab-1');
+
+  const selectedTabIdInstall = (id: string) => {
```
Comment (tetiana-ivanchenko) [inline src/App.tsx:16]: one more time - name should describe what fucntion do, why install?

File: src/App.tsx [hunk 6]
```diff
@@ -10,31 +11,31 @@ export const tabs = [
...
+    setSelectedTabId(id);
+  };
+
+  const tabSelect = tabs.find(tab => tab.id === selectedTabId);
```
Comment (tetiana-ivanchenko) [inline src/App.tsx:20]:
```suggestion
  const selectedTab = tabs.find(tab => tab.id === selectedTabId);
```

**Review by tetiana-ivanchenko — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 7]
```diff
@@ -10,31 +11,31 @@ export const tabs = [
 ];
 
 export const App: React.FC = () => {
+  const [selectedTabId, setSelectedTabId] = useState('tab-1');
```
Comment (sTorba24) [inline src/App.tsx:14]:
```suggestion
  const [selectedTabId, setSelectedTabId] = useState(tabs[0].id);
```

File: src/App.tsx [hunk 8]
```diff
@@ -10,31 +11,31 @@ export const tabs = [
...
+
+  const IdSelectedTab = (id: string) => {
+    setSelectedTabId(id);
+  };
```
Comment (sTorba24) [inline src/App.tsx:18]: you can name it like `handleTabSelected`
Also don't use naming from uppercase
`idSelectedTab` doesn't sound like name describing that it is function

File: src/App.tsx [hunk 9]
```diff
@@ -10,31 +11,31 @@ export const tabs = [
...
+
+  const selectTab = tabs.find(tab => tab.id === selectedTabId);
+
   return (
     <div className="section">
       <h1 className="title">
-        Selected tab is Tab 1
+        Selected tab is
+        {' '}
+        {selectTab?.title}
```
Comment (sTorba24) [inline src/App.tsx:27]: use interpolation here

File: src/App.tsx [hunk 10]
```diff
@@ -10,31 +11,31 @@ export const tabs = [
...
-            </li>
-
-            <li data-cy="Tab">
-              <a href="#tab-3" data-cy="TabLink">Tab 3</a>
-            </li>
-          </ul>
+          <Tabs
+            tabs={tabs}
+            selectedTabId={selectedTabId}
+            getSelectedTabId={IdSelectedTab}
```
Comment (sTorba24) [inline src/App.tsx:34]: it's not get function
if you will use it for onClick so name it like `onTabSelected`

File: src/components/Tabs/Tabs.tsx [hunk 11]
```diff
@@ -1 +1,42 @@
-export const Tabs = () => {};
+import React from 'react';
+import ClassNames from 'classnames';
+
+type Props = {
+  tabs: {
+    id: string;
+    title: string;
+    content: string;
+  }[]
```
Comment (sTorba24) [inline src/components/Tabs/Tabs.tsx:9]: create Tab interface to describe this object

File: src/components/Tabs/Tabs.tsx [hunk 12]
```diff
@@ -1 +1,42 @@
...
+  selectedTabId: string;
+  getSelectedTabId: (id: string) => void;
+};
+
+export const Tabs: React.FC<Props> = (
+  { tabs, selectedTabId, getSelectedTabId },
+) => {
```
Comment (sTorba24) [inline src/components/Tabs/Tabs.tsx:16]:
```suggestion
export const Tabs: React.FC<Props> = ({ 
  tabs, 
  selectedTabId, 
  getSelectedTabId 
}) => {
```

File: src/components/Tabs/Tabs.tsx [hunk 13]
```diff
@@ -1 +1,42 @@
...
+  return (
+    <ul>
+      {
+        tabs.map(tab => {
+          const { id, title } = tab;
+          const comparisonId = id === selectedTabId;
```
Comment (sTorba24) [inline src/components/Tabs/Tabs.tsx:22]:
```suggestion
          const isActive= id === selectedTabId;
```

**Review by sTorba24 — CHANGES_REQUESTED** [review]
Good work! Let's improve your code ;)

**Review by AlisaVlasova — APPROVED** [review]

**Pull request closed without merging**